userguide/one_shot_object_detection/README.md

```diff
@@ -0,0 +1,120 @@
+# One-Shot Object Detection
+
+One-Shot object detection is the task of learning to detect from as few as one example per category. Unlike the Object Detector which requires many varied examples of objects in the real world, the One-Shot detector only requires a single canonical example of an object to train the classifier on and performs predictions for that category in the wild. It is best suited for 2D objects or when there isn't much expected variance in the examples, such as playing cards, logos, road signs and clapperboards. One-Shot detector is not suitable for 3D objects such as faces, animals and cars. 
+
+Given an image and a one-shot detector trained on category **stop sign**, the output prediction will look like:
+
+![Stop Sign vector and prediction images](images/stop_sign_sample.jpg) 
+
+
+#### Introductory Example
+
+In this example, the goal is to **localize** where in the image instances of the category **stop sign** are present. For this task we supply a single image of a stop sign as a starter image:
+
+
+```python
+import turicreate as tc
+
+# Load the starter images
+starter_images = tc.SFrame({'image':[tc.Image('stop_sign.png')],
+                   'label':['stop_sign']})
+
+# Load test images
+test_data = tc.SFrame({'image':[tc.Image('stop_sign_photo1.png'), 
+                                tc.Image('stop_sign_photo2.png')],
+                        'label':['stop_sign', 'stop_sign']})
+
+# Create a model. This step may take several hours.                                      
+model = tc.one_shot_object_detector.create(starter_images, 'label')
+
+# Save predictions on the test set as a column in the SFrame
+test_data['predictions'] = model.predict(test_data)
+
+# Draw prediction bounding boxes on the test images
+test_data['image_pred'] = tc.object_detector.util.draw_bounding_boxes(test_data['image'], test_data['predictions']) 
+
+# Evaluate the model and save the results in a dictionary
+metrics = model.evaluate(test_data)
+
+# Save the model for later use in TuriCreate
+model.save('mymodel.model')
+
+# Export for use in Core ML
+model.export_coreml('MyCustomOneShotDetector.mlmodel')
+```
+
+Previewing the synthetic images generated for training is useful to gauge the kind of test images the model would be trained to handle. We can preview the augmented images generated and used by the model for training as follows:  
```

Comment:
Let's use either "synthetic images" or "augmented images". Using both may confuse readers. Ideally, we should stick to "synthetic" throughout since our API calls also have "synthetic".

How about, "Before starting the training process, you can preview the synthetic images the toolkit generates by calling the `preview_synthetic_training_data` utility. This helps you understand data the model would be trained on."

Reply:
This would also apply to the names of the files as well. You use augmented and synthetic exchangeably